plugins/jobtype/src/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -0,0 +1,255 @@
+      e.printStackTrace();
+      throw new HadoopSecurityManagerException("Failed to create the token file.", e);
+    }
+
+    hadoopSecurityManager.prefetchToken(tokenFile, props, log);
+
+    return tokenFile;
+  }
+
+  /**
+   * <pre>
+   * If there's a * specification in the "jar" argument (e.g. jar=./lib/*,./lib2/*),
+   * this method helps to resolve the * into actual jar names inside the folder, and in order.
+   * This is due to the requirement that Spark 14 doesn't seem to do the resolution for users
+   * 
+   * TODO: test case required: one replacement, two replacement.
+   * </pre>
+   * 
+   * @param unresolvedJarSpec
+   * @return jar file list, comma separated, all .../* expanded into actual jar names in order
+   * 
+   */
+  public static String resolveWildCardForJarSpec(String workingDirectory, String unresolvedJarSpec,
+          Logger log) {
+
+    log.debug("resolveWildCardForJarSpec: unresolved jar specification: " + unresolvedJarSpec);
+
+    if (unresolvedJarSpec == null || unresolvedJarSpec.isEmpty())
+      return null;
+
+    StringBuilder resolvedJarSpec = new StringBuilder();
+
+    String[] unresolvedJarSpecList = unresolvedJarSpec.split(",");
+    for (String s : unresolvedJarSpecList) {
+      // if need resolution
+      if (s.endsWith("*")) {
+        String fileName = String.format("%s/%s", workingDirectory, s.substring(0, s.length() - 2));
+        String[] jarList = new File(fileName).list(jarFilter);
+        for (String ls : jarList) {
+          String lsFilename = fileName + "/" + ls + ",";
+          resolvedJarSpec.append(lsFilename);
+        }
+      } else { // no need for resolution
+        resolvedJarSpec.append(s + ",");
+      }
+    }
+
+    log.debug("resolveWildCardForJarSpec: resolvedJarSpec: " + resolvedJarSpec);
+
+    // remove the trailing comma
+    int lastCharIndex = resolvedJarSpec.length() - 1;
+    if (lastCharIndex >= 0 && resolvedJarSpec.charAt(lastCharIndex) == ',') {
+      resolvedJarSpec.deleteCharAt(lastCharIndex);
+    }
+
+    return resolvedJarSpec.toString();
+  }
+
+  /**
+   * <pre>
+   * This method looks for the proper user execution jar.  
+   * The user input is expected in the following 2 formats:
+   *   1. ./lib/abc
+   *   2. ./lib/abc.jar
+   * 
+   * This method will use prefix matching to find any jar that is the form of abc*.jar,
+   * so that users can bump jar versions without doing modifications to their Hadoop DSL.
+   * 
+   * This method will throw an Exception if more than one jar that matches the prefix is found
+   * 
+   * @param workingDirectory
+   * @param userSpecifiedJarName
+   * @return the resolved actual jar file name to execute
+   */
+  public static String resolveExecutionJarName(String workingDirectory,
+          String userSpecifiedJarName, Logger log) {
+
+    log.debug("Resolving execution jar name: working directory: " + workingDirectory
+            + ", user specified name: " + userSpecifiedJarName);
```

Comment:
Interesting point.

The two public methods will still exist as the input value and return value that the methods interface with are slightly different.  However, I believe there's a core of maybe ~15 lines that can probably be extracted out.